### PR TITLE
Allow package downgrade in "build_dependencies:install" task

### PR DESCRIPTION
**Pull request by lslezak**
- We get this error in GitHub Actions:
  ```
  Problem: the to be installed skelcd-control-openSUSE-20210224-lp153.1.2.x86_64 requires 'yast2-kdump',
  but this requirement cannot be provided
    not installable providers: yast2-kdump-4.3.3-1.32.x86_64[repo-oss]
                     yast2-kdump-4.3.4-lp153.1.9.x86_64[yast]
   Solution 1: Following actions will be done:
    downgrade of systemd-246.10-1.8.x86_64 to systemd-246.9-1.9.x86_64
    downgrade of udev-246.10-1.8.x86_64 to udev-246.9-1.9.x86_64
  ```
- https://github.com/yast/skelcd-control-MicroOS/runs/1986157503
- The problem is that OBS builds the Docker image against the latest packages, it uses the latest packages which are not published yet (waiting in the queue). If you later want to install additional packages the OSS repository still contains older version and this might cause a zypper error.
- When the task is used locally (from a terminal) then it asks for confirmation, so the user should see possible downgrades and potentially stop the task. So it should be safe in manual case.
- https://bugzilla.suse.com/show_bug.cgi?id=1182795
- 1.4.15

### CODE REVIEW

File: lib/tasks/build_dependencies.rake [hunk 1]
```diff
@@ -56,7 +56,9 @@ namespace :build_dependencies do
     else
       sudo = Process.euid.zero? ? "" : "sudo"
       interactive = $stdin.tty? ? "" : "--non-interactive"
-      cmd = "#{sudo} zypper #{interactive} install #{escaped_list}"
+      # allow package downgrade to avoid failures in CI when the installed
+      # packages are higher than the available ones
+      cmd = "#{sudo} zypper #{interactive} install --allow-downgrade #{escaped_list}"
```
Comment (jreidinger) [inline lib/tasks/build_dependencies.rake:61]: Do we really need downgrade? It can cause failures as it will use older packages then it can.

Comment (lslezak) [reply lib/tasks/build_dependencies.rake:61]: I do not like downgrade as well, but the "older" packages are actually not old, they are in the latest published version. At least when running in CI. When running locally user needs to confirm the downgrade manually so this should be pretty safe.

Alternatively we could do this in CI:

```
# sync the installed packages with the OSS repo
zypper dup
# install build dependencies
rake build_dependencies:install
```

The disadvantage is that we would need to touch all places which use `rake build_dependencies:install`. But IIRC that should be just few places...

Comment (lslezak) [reply lib/tasks/build_dependencies.rake:61]: My quick grep found 23 affected packages...

Comment (jreidinger) [reply lib/tasks/build_dependencies.rake:61]: ok, so lets allow for now that allow downgrade

**Review by jreidinger — APPROVED** [review]

**Pull request merged**